versions-common/src/main/java/org/codehaus/mojo/versions/api/DefaultVersionsHelper.java

```diff
@@ -868,19 +686,172 @@ else if ( !excludePropertiesList.isEmpty() && excludePropertiesList.contains( pr
+        {
+            String baseUri;
+            URI fileUri;
+
+            private RulesUri( String baseUri, URI fileUri )
+            {
+                this.baseUri = baseUri;
+                this.fileUri = fileUri;
+            }
+
+            static RulesUri build( String rulesUri ) throws URISyntaxException
+            {
+                int split = rulesUri.lastIndexOf( '/' );
+                return split == -1
+                        ? new RulesUri( rulesUri, new URI( "" ) )
+                        : new RulesUri( rulesUri.substring( 0, split ) + '/',
+                        new URI( split + 1 < rulesUri.length()
+                                ? rulesUri.substring( split + 1 )
+                                : "" ) );
+            }
+        }
+
+
+        private RuleSet getRulesUsingTransporter()
+        {
+            RulesUri uri;
+            try
+            {
+                uri = RulesUri.build( rulesUri );
+            }
+            catch ( URISyntaxException e )
+            {
+                log.warn( "Invalid rulesUri protocol: " + e.getMessage() );
+                return null;
+            }
+
+            RemoteRepository repository = new RemoteRepository.Builder( serverId, null, uri.baseUri )
+                    .build();
+            return transporterFactoryMap
+                    .values()
+                    .stream()
+                    // highest priority first -> reversing the order of arguments:
+                    .sorted( ( f1, f2 ) -> Float.compare( f2.getPriority(), f1.getPriority() ) )
+                    .map( factory ->
+                    {
+                        try
+                        {
+                            return factory.newInstance( mavenSession.getRepositorySession(), repository );
+                        }
+                        catch ( NoTransporterException e )
+                        {
+                            log.warn( "No transporter possible for " + uri.baseUri + ": "
+                                    + e.getMessage() );
+                            return null;
+                        }
+                    } )
+                    .filter( Objects::nonNull )
+                    .map( transporter ->
+                    {
+                        try
+                        {
+                            GetTask getTask = new GetTask( uri.fileUri );
+                            transporter.get( getTask );
+                            return new RuleXpp3Reader().read( new StringReader( getTask.getDataString() ) );
+                        }
+                        catch ( Exception e )
+                        {
+                            log.warn( "Error while reading the rules string: " + e.getMessage() );
+                            return null;
+                        }
+                    } )
+                    .filter( Objects::nonNull )
+                    .findFirst()
+                    .orElse( null );
+        }
```

Comment:
I'm not sure if we can use Resolver Transports in such way.
Rather it is for internal use by Resolver.

In this case - we simply need download a resource from remote url.
Better will be use eg HttpClient from HttpComponents.

Reply:
Hmm, are you sure? If that is the case, then I'll provably need to use different transports depending on protocol (file, classpath, etc).

Reply:
Hmm, are you sure? If that is the case, then I'll provably need to use different transports depending on protocol (file, classpath, etc).

Reply:
@cstamas - can have a best knowledge about it.

Reply:
If not, HttpClient5 is lightweight enough and I'd like to use the Fluent flavour of it, if you don't mind.

Reply:
Posted a question to dev@maven.apache.org, but I can't see it. Do I need an invite?

Reply:
HttpClient5 will be ok.

I would like to introduce next util class for getting resource from remote, with method like:

```
 ... downolad( uri, serverId, mavenSession ) 
```

in method we can find proper server configuration from `mavenSession.settings`
Maven settings can also has a proxy configuration which should be under consideration

Reply:
I was thinking of something like that actually. E.g. provide several implementations per protocol (classpath, file, http/https) and use a factory method to select the matching implementation from a map.

Or use Wagon without WagonManager instead? Since Wagon basically already provides such a tnin layer, currently using HttpClient 4.5. Is Wagon going to be retired altogether?

Reply:
@hboutemy Do you know if we can continue to use Wagon or is it going to be retired and we should be using e.g. HttpClient directly?

Reply:
> Posted a question to [dev@maven.apache.org](mailto:dev@maven.apache.org), but I can't see it. Do I need an invite?

You should first subscribe to list.

Reply:
? should works .... after send email to: dev-subscribe at maven.apache.org